### DISCUSSION TITLE
Shell broken

### DISCUSSION THREAD

**Discussion by Mitony07** [ohmyzsh/ohmyzsh#12554] [Help]
Hello,

i installed zsh with `apt install zsh` and then used `sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"` to install omz. I replaced the .zshrc file with the one on my other sytem. As soon as i type a command the shell dosnt know it gets broken. Any ideas?

![grafik](https://github.com/user-attachments/assets/cc77e469-e574-46f7-8f53-b149d6f9d94c)
![grafik](https://github.com/user-attachments/assets/686aa8df-baa5-4fe4-929e-b06e7b912561)

**Comment by mcornella** [comment]
You should provide more information to diagnose this. With the info provided, I understand this:

1. When you type, what you typed gets merged with the prompt.
2. You don't have the `lla` alias available.

I'd say (1) is probably caused by using a non-UTF8 locale, so zsh can't accurately tell how many characters have been written to the terminal. You can check this out by running `locale`, you should have one that ends in `.utf8` or `.UTF-8`. See https://github.com/ohmyzsh/ohmyzsh/wiki/FAQ#how-do-i-change-my-locale.

(2) might happen if your current zshrc file does not contain the aliases that you wanted, either because it's badly copied or something is commented out, or any other possibility. There should be a line in there saying `alias lla=...` and no lines saying `unalias lla`.

Without more info it's impossible to confirm any of that. If this doesn't help, please post your redacted .zshrc file.

**Comment by KianAnbarestani** [comment]
Likely your new` .zshrc` (from the other system) references aliases or plugins that don’t exist on this machine. Try these steps:

1. Check that `source $ZSH/oh-my-zsh.sh` is still in your `~/.zshrc.`
2. Inspect `plugins=(...)` in your `.zshrc` to ensure all plugins are installed/available.
3. Verify your `$PATH` (run `echo $PATH`) includes standard bin folders.
4. Restart the shell (`exec zsh`) to reload. If a command still fails, see if it’s defined by an alias or plugin that’s missing.
